### PR TITLE
Reader Stream Refresh: add photo-only card

### PR DESCRIPTION
**Pull request by jancavan**
Fixes https://github.com/Automattic/wp-calypso/issues/8518

1af553b4-8bc4-11e6-8329-e839005b2279

### CODE REVIEW

**Comment by bluefuton** [comment]
This is looking awesome @jancavan! A few selected screenshots:

screen shot 2016-10-19 at 15 32 49
screen shot 2016-10-19 at 15 32 59
screen shot 2016-10-19 at 15 33 16

cc @fraying

**Comment by bluefuton** [comment]
One minor issue is that I sometimes see a flash of the standard card whilst the photo-only card is loading. It's because the card hasn't received the `display_type` in time for the first render. Perhaps we can do that earlier @blowery?

Video: https://cloudup.com/cldHLUdue13

**Comment by fraying** [comment]
We're going to be hiding the excerpts, right?

**Comment by jancavan** [comment]
@bluefuton As mentioned here: https://github.com/Automattic/wp-calypso/pull/8823#issuecomment-254644394

> If the post has 1 or more images and less than 130 characters: Photo card (using featured image or first image).

**Comment by bluefuton** [comment]
> We're going to be hiding the excerpts, right?

Ah yes, we are indeed :) As of b995d5c it's hidden and they look like this:

screen shot 2016-10-19 at 20 36 31

**Comment by blowery** [comment]
Changed up the background to use a linear gradient instead of the full overlay, which makes many cards look a fair bit better. No more muted colors! :)

Also expanded the title. It was getting chopped at 50 characters, which felt way too short. Instead, dynamically truncate it using overflow and text-overflow along with an ellipsis on supported browsers.

**Comment by jancavan** [comment]
Thanks @blowery 👏  

In https://github.com/Automattic/wp-calypso/pull/8843/commits/a3ad9755b7640d562a5e914d8fdff0ee0a7eeb6a, made a minor adjustment to the linear gradient overlay so it doesn't end abruptly (only obvious in lighter backgrounds).

**Before:**
screenshot 2016-10-19 09 56 11

**After:**
screenshot 2016-10-19 09 56 53

**Comment by blowery** [comment]
@jancavan woo, that's looking nice!

File: client/blocks/reader-post-card/index.jsx [hunk 1]
```diff
@@ -96,7 +96,7 @@ export default class RefreshPostCard extends React.Component {
 		const featuredImage = post.canonical_image;
 		const isPhotoOnly = post.display_type & DisplayTypes.PHOTO_ONLY;
 		const title = truncate( post.title, {
-			length: isPhotoOnly ? 50 : 140,
+			length: isPhotoOnly ? 500 : 140,
```
Comment (blowery) [inline client/blocks/reader-post-card/index.jsx]: do we even need this now?

**Comment by jancavan** [comment]
> One minor issue is that I sometimes see a flash of the standard card whilst the photo-only card is loading.

I noticed this too. The excerpts have completely been removed in photo cards, but the post title still flashes using the default card's style: https://cloudup.com/cTNFfmXaXTc

Example: http://calypso.dev:3000/read/feeds/23059853 (first post in stream)

**Comment by blowery** [comment]
@jancavan the font color was needed because the ellipsis is black otherwise...

**Comment by jancavan** [comment]
@blowery My bad. Added it back: https://github.com/Automattic/wp-calypso/pull/8843/commits/915caeccdecd1fa55de838d1e2f5b51993677bb9

**Comment by blowery** [comment]
I think the loading issue can be handled in a separate PR. We'll need to assign PHOTO_ONLY earlier i think.

**Comment by bluefuton** [comment]
These look lovely. Let's 🚢 

screen shot 2016-10-20 at 11 10 48

**Pull request merged**